src-v2/NTS.Domain.Core/Objects/Start.cs

```diff
@@ -10,39 +12,34 @@ public Start(
         int loopNumber,
         double distance,
         double totalDistance,
-        Timestamp startAt
+        TimeSpan startAt
     )
     {
         Athlete = athlete;
         Number = number;
         PhaseNumber = loopNumber;
         Distance = distance;
-        TotalDistance = totalDistance.FloorWholeNumberToTens();
+        TotalDistance = totalDistance.RoundWholeNumberToTens();
         StartAt = startAt;
     }
 
+    public Start(Participation participation)
+    {
+        Athlete = participation.Combination.Name;
+        Number = participation.Combination.Number;
+        var phaseIndex = participation.Phases.IndexOf(participation.Phases.Current);
+        PhaseNumber = phaseIndex + 2;
```

Comment:
This is a classic case of magic string usage being hard to determine. After some pondering I've deduced that you're selecting the Phase after `Current`. Are you sure this is behaving correctly with the change you've made in `Participation.cs`? Isn't `Current` going to be the phase you need here, rather than the next one?

From a quality of code standpoint I shouldn't have to deduce things like that, it should be plain text to read. We do have a `NumberOf` extension which returns the 1-base index of an item in a collection

Reply:
Surprisingly, even with the change in `Participation.cs` the Current phase is not changed by `StartIfNext` it only sets the StartTime of the next phase, the Current phase is changed by SelectNext which is executed 30 minutes after the StartTime of the previous one in the following Phase logic:
```
  var notProcessingWindow = TimeSpan.FromMinutes(30); // TODO settings: use settings?
  if (isComplete && snapshot.Timestamp > Current.GetOutTime() + notProcessingWindow)
  {
      SelectNext();
  }
 ```